pkg/provisioner/ironic/ironic.go

```diff
@@ -1212,6 +1245,23 @@ func (p *ironicProvisioner) Provision(hostConf provisioner.HostConfigData) (resu
 		return transientError(provisioner.NeedsRegistration)
 	}
 
+	if p.host.Spec.AutomatedCleaningMode == metal3v1alpha1.CleaningModeDisabled {
+		p.log.Info("automated cleaning is disabled for host",
+			"ID", ironicNode.UUID,
+			"automated_clean", ironicNode.AutomatedClean,
+		)
+		p.publisher("CleaningDisabled", "Automated cleaning disabled")
+	}
+
+	if ironicNode.AutomatedClean != nil {
+		if (p.host.Spec.AutomatedCleaningMode == metal3v1alpha1.CleaningModeDisabled && *ironicNode.AutomatedClean == true) || (p.host.Spec.AutomatedCleaningMode != metal3v1alpha1.CleaningModeDisabled && *ironicNode.AutomatedClean == false) {
```

Comment:
Maybe move this check inside `updateAutomatedClean` to avoid duplication of logic?